save-backend/src/main/kotlin/org/cqfn/save/backend/postprocessor/DockerSecretsDatabaseProcessor.kt

```diff
@@ -14,11 +13,18 @@ import java.util.Properties
 /**
  * Post processor that collects credentials for database from docker secrets
  */
-@Profile("docker-secrets")
-class DockerSecretsDatabaseProcessor : EnvironmentPostProcessor {
-    private val log = LoggerFactory.getLogger(DockerSecretsDatabaseProcessor::class.java)
+class DockerSecretsDatabaseProcessor(
+    logFactory: DeferredLogFactory
+) : EnvironmentPostProcessor {
+    private val log = logFactory.getLog(DockerSecretsDatabaseProcessor::class.java)
 
     override fun postProcessEnvironment(environment: ConfigurableEnvironment, application: SpringApplication) {
+        // Since `EnvironmentPostProcessor` is created before application context is fully initialized, usual means of registration control
+        // like `@Profile` cannot be used. `environment` here should contain at least profiles set from env variable or system properties.
```

Comment:
so, previously, the annotation didn't actually worked?

Reply:
Yes, neither annotation nor logging